course02/homework02/danielellis/Buffer.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <deque>
+#include <chrono>
+#include <vector>
+#include <stdexcept>
+
+struct WindowConfig
+{
+	std::chrono::seconds mDuration = std::chrono::seconds(10);
+	int mMaxNumMessages = 5;
+};
+
+template<class Message>
+struct TimestampedMessage
+{
+	std::chrono::time_point<std::chrono::steady_clock> mTimestamp;
+	Message mMessage;
+};
+
+template<class Message>
+class Buffer
```

Comment:
I think that this Buffer is effectively a sliding window buffer, and the other Sliding Window class is the per user throttler